zerver/tests/test_narrow.py

```diff
@@ -925,6 +926,16 @@ def test_get_messages_with_search_pgroonga(self):
         self.assertEqual(multi_search_result['messages'][0]['match_content'],
                          '<p><span class="highlight">Can</span> you <span class="highlight">speak</span> <a href="https://en.wikipedia.org/wiki/Japanese" target="_blank" title="https://en.wikipedia.org/wiki/Japanese">https://en.<span class="highlight">wiki</span>pedia.org/<span class="highlight">wiki</span>/Japanese</a>?</p>')
 
+        narrow = [dict(operator='search', operand='https://google.com')]
+        link_search_result = self.get_and_check_messages(dict(
+            narrow=ujson.dumps(narrow),
+            anchor=0,
+            num_after=10,
+        ))  # type: Dict[str, Dict]
+        self.assertEqual(len(link_search_result['messages']), 1)
+        self.assertEqual(link_search_result['messages'][0]['match_content'],
+                         '<p><a href="https://google.com" target="_blank" title="https://google.com"><span class="highlight">https://google.com</span></a></p>')
+
```

Comment:
This test will only be run against the pgroonga backend; we probably want a similar test with `USING_PGROONGA=False`.  Might be worth thinking about tweaking the framework here to support the same code being used for both, since I think both search backends should do this correctly.